tools/toolchains/mbed_toolchain.py

```diff
@@ -894,63 +894,56 @@ def _add_all_regions(self, region_list, active_region_name):
     def add_regions(self):
         """Add regions to the build profile, if there are any.
         """
+
+        if not getattr(self.target, "bootloader_supported", False):
```

Comment:
I would rather encode the default behaviour in `self.target` rather than define the default here, other places where this is used may assume a different default. There is also a danger of typos using this technique that would not cause exceptions as if there is a typos the result is always `False`. Maybe these are relatively minor points but they do somewhat hinder maintainability so should be avoided in code we wish to maintain in the long term.

Reply:
This is how it is done in regions().
 It should be done properly but I don't have entire picture how config works yet . I found this the easiest way to fix now and later to even rewrite this add_regions to catch errors better.

Do you suggest to have target method to return if bootloader is supported and add_regions() would not even be called if not?